src/drivers/src/stm32f4xx.c

```diff
@@ -38,7 +39,7 @@ HAL_StatusTypeDef CANlib_TransmitFrame(Frame *frame, CANlib_Bus_T bus) {
   return HAL_CAN_AddTxMessage(hcan, &pHeader, frame->data, &pTxMailbox);
 }
 
-void CANlib_ReadFrame(Frame *frame, CANlib_Bus_T bus) {
+bool CANlib_ReadFrame(Frame *frame, CANlib_Bus_T bus) {
```

Comment:
If it returns anything it should probably be HAL_StatusTypeDef.

Reply:
The thing is `HAL_CAN_GetRxMessage` only returns `HAL_OK` or `HAL_ERROR` anyway; and also it could possibly not even be called if both FIFO's are empty